Libraries/LibJS/Bytecode/Interpreter.cpp

```diff
@@ -355,353 +355,237 @@ FLATTEN_ON_CLANG void Interpreter::run_bytecode(size_t entry_point)
     size_t& program_counter = running_execution_context.program_counter;
     program_counter = entry_point;
 
-    // Declare a lookup table for computed goto with each of the `handle_*` labels
-    // to avoid the overhead of a switch statement.
-    // This is a GCC extension, but it's also supported by Clang.
+    auto& next_instruction = *reinterpret_cast<Instruction const*>(&bytecode[program_counter]);
+    auto fn = Interpreter::dispatch_instruction_table[static_cast<size_t>(next_instruction.type())];
```

Comment:
Maybe experiment with putting these pointers into the bytecode stream? this lookup is probably the single most mispredicted thing in your code.